src/GoodsOutNote.php

```diff
@@ -116,6 +117,15 @@ public function toJson(): array
         ];
     }
 
+    /**
+     * @param ModelInterface $goodsOutNoteToCompare
+     * @return bool
+     */
+    public function equals(ModelInterface $goodsOutNoteToCompare): bool
+    {
+        return $this->toJson() === $goodsOutNoteToCompare->toJson();
```

Comment:
@ampdan this is not correct. if you pass two arrays using different orders of the keys it will say that is not equals even though it is. also you are not checking the type.. if I pass any other model that implements same interface and has same values it will say that is equals and it is not. 

this also needs fixing.. 
https://github.com/snowio/brightpearl-data-model/blob/269959f07a9590803deb870eaaa9992aa1a65d64/src/Contact.php#L124-L189

we have quite a few datamodels, use them as reference please.

Reply:
Follow this pattern please https://github.com/snowio/orderhive-data-model/blob/ea62ea45d51c1953cdb50656f3d549fb826bd830/src/Product/Remark.php#L32-L37